swagger_server/itm/itm_scenario_session.py

```diff
@@ -712,15 +712,22 @@ def lookup_probe_response(self, action: Action) -> ProbeResponse:
 
         Args:
             body: The probe response body as a dict.
+        Returns:
+            The probe corresponding to the action, or None if there is no corresponding probe.
         """
+
+        if action.action_id == None or action.action_id == '':
+            return None
+
         currentProbe = self.current_isso.probe_system.probe_yamls[self.current_isso.probe_system.current_probe_index]
-        choice_id = ""
+        choice_id = None
         # need to go back through to find the choice from probeYamlOption (not stored in action)
         for option in currentProbe.options:
```

Comment:
can replace for loop with 
`choice_id = next((option.ta1_id for option in currentProbe.options if option.assoc_action["action_id"] == action.action_id), None)`
matches choice_id to the first option with the same action_id as action.action_id. If no match, default to none

Reply:
This is a case where I'd choose readability over replacing a loop with one line, especially as I may be the one maintaining it.  It's part of a small function, so it's not helping to condense a large block of code, either.  Also, this wasn't code I added for the ticket.  I realize your approach is more Pythonesque.  Does @brianpippin  have an opinion or preference?

Reply:
This is more a coding style choice and I don't want to be dictating anything there, so long as it is readable and understandable, I am good with either option.